include/nbl/asset/IImageView.h

```diff
@@ -108,17 +108,35 @@ class IImageView : public IDescriptor
 				return false;
 			*/
 
-			// declared usages that are not a subset
-			if (!imgParams.usage.hasFlags(_params.subUsages))
+			const auto kValidUsages = core::bitflag(IGPUImage::EUF_SAMPLED_BIT)|IGPUImage::EUF_STORAGE_BIT|
+				IGPUImage::EUF_COLOR_ATTACHMENT_BIT|IGPUImage::EUF_DEPTH_STENCIL_ATTACHMENT_BIT|IGPUImage::EUF_TRANSIENT_ATTACHMENT_BIT|IGPUImage::EUF_INPUT_ATTACHMENT_BIT|
+				IGPUImage::EUF_SHADING_RATE_IMAGE_BIT_NV|IGPUImage::EUF_FRAGMENT_DENSITY_MAP_BIT_EXT;
+			// https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkImageViewCreateInfo.html#VUID-VkImageViewCreateInfo-image-04441
+			if ((imgParams.usage.value&kValidUsages)==0u)
 				return false;
 
+			const auto& subresourceRange = _params.subresourceRange;
+			// declared some usages but they are not a subset
+			{
+				// https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkImageViewCreateInfo.html#VUID-VkImageViewCreateInfo-pNext-02663
+				if (subresourceRange.aspect.hasFlags(IImage::EAF_STENCIL_BIT) && !imgParams.stencilUsage.hasFlags(_params.subUsages))
+					return false;
+				// https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkImageViewCreateInfo.html#VUID-VkImageViewCreateInfo-pNext-02664
+				if ((subresourceRange.aspect&(~IImage::EAF_STENCIL_BIT)).value && !imgParams.usage.hasFlags(_params.subUsages))
+					return false;
+			}
+
+			// https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkImageViewCreateInfo.html#VUID-VkImageViewCreateInfo-image-02087
+			if (_params.format!=asset::EF_R8_UINT)
```

Comment:
wait What?! am I seeing this correclty? you return false on formats not equal to R8_UINT
am I missing something here?

based on the VUID you should also check for `VK_IMAGE_USAGE_SHADING_RATE_IMAGE_BIT_NV` usage and `[shadingRateImage](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#features-shadingRateImage` feature enabled